hyperdrive.go

```diff
@@ -48,29 +42,22 @@ func New(signer sig.SignerVerifier, dispatcher replica.Dispatcher) Hyperdrive {
 		dispatcher: dispatcher,
 
 		shardReplicas: map[sig.Hash]replica.Replica{},
-
-		ticksPerShard: map[sig.Hash]int{},
 	}
 }
 
 func (hyperdrive *hyperdrive) AcceptTick(t time.Time) {
 	// 1. Increment number of ticks seen by each shard
 	for shardHash := range hyperdrive.shardReplicas {
-		hyperdrive.ticksPerShard[shardHash]++
-
-		if hyperdrive.ticksPerShard[shardHash] > NumTicksToTriggerTimeOut {
-			// 2. Send a TimedOut transition to the shard
-			if replica, ok := hyperdrive.shardReplicas[shardHash]; ok {
-				replica.Transition(state.TimedOut{Time: t})
-				hyperdrive.ticksPerShard[shardHash] = 0 // Reset tickPerShard
-			}
+
+		// 2. Send a Ticked transition to the shard
+		if replica, ok := hyperdrive.shardReplicas[shardHash]; ok {
+			replica.Transition(state.Ticked{Time: t})
```

Comment:
Why did we remove the tick counter? It makes sure that we can guarantee a particular range for the duration between ticks.

Reply:
Handling ticks has been moved to the StateMachine.